hphp/runtime/vm/jit/smashable-instr-arm.h

```diff
@@ -77,6 +77,22 @@ constexpr size_t kSmashJccTargetOff = 12;
 
 ///////////////////////////////////////////////////////////////////////////////
 
+template<typename T>
+  static void patchInstr(TCA inst, T target) {
+  *reinterpret_cast<T*>(inst) = target;
+  auto const end = reinterpret_cast<char*>(inst + sizeof(T));
+  auto const begin = end - sizeof(T);
+  __builtin___clear_cache(begin, end);
+}
+
+using TCA32 = uint32_t;
```

Comment:
This is pretty weird, since `TCA` is a pointer type.  Can you just use `uint32_t` directly?  Many of the places you're using this type call something generic like `patchInstr()` with it, which really just expects any 32-bit immediate.